Google sign-in failing for some geolocations due to Google Identity Services (GSI) client issues

### Describe the bug

I've been using Metabase and Google auth without issues, and suddenly for users in Brazil, instead of the authentication page, we see the "Something's gone wrong page." Users in other locations, such as Portugal and USA, can log in to Metabase normally. Please take a look at the screencast below.

### To Reproduce

1. Metabase instance with Google auth enabled
2. Enable VPN going through Sao Paulo/Brazil region
3. Open Metabase and get redirected to the login page

![image](https://github.com/metabase/metabase/assets/6169152/58a3ce0f-1feb-46d3-b5b5-a0879339f8b6)

### Expected behavior

Proposal is to stop embedding Google Auth and redirect to Google Auth page (something like OpenAI does with chatgpt login) as it seems more stable

### Logs

```
api.js:209     GET https://metabase.<domain>/api/user/current 401
(anonymous) @ api.js:209
value @ api.js:167
value @ api.js:162
(anonymous) @ api.js:135
u @ runtime.js:63
(anonymous) @ runtime.js:293
(anonymous) @ runtime.js:118
p @ app-main.bundle.js?28f670131887d2ef460e:2
a @ app-main.bundle.js?28f670131887d2ef460e:2
(anonymous) @ app-main.bundle.js?28f670131887d2ef460e:2
(anonymous) @ app-main.bundle.js?28f670131887d2ef460e:2
(anonymous) @ api.js:32
(anonymous) @ api.js:111
u @ runtime.js:63
(anonymous) @ runtime.js:293
(anonymous) @ runtime.js:118
p @ app-main.bundle.js?28f670131887d2ef460e:2
a @ app-main.bundle.js?28f670131887d2ef460e:2
(anonymous) @ app-main.bundle.js?28f670131887d2ef460e:2
(anonymous) @ app-main.bundle.js?28f670131887d2ef460e:2
(anonymous) @ api.js:60
(anonymous) @ user.js:13
r @ createAction.js:16
u @ createAction.js:22
(anonymous) @ user.js:24
u @ runtime.js:63
(anonymous) @ runtime.js:293
(anonymous) @ runtime.js:118
f @ app-main.bundle.js?28f670131887d2ef460e:2
a @ app-main.bundle.js?28f670131887d2ef460e:2
(anonymous) @ app-main.bundle.js?28f670131887d2ef460e:2
(anonymous) @ app-main.bundle.js?28f670131887d2ef460e:2
(anonymous) @ user.js:22
(anonymous) @ redux.js:204
u @ runtime.js:63
(anonymous) @ runtime.js:293
(anonymous) @ runtime.js:118
y @ app-main.bundle.js?28f670131887d2ef460e:2
a @ app-main.bundle.js?28f670131887d2ef460e:2
(anonymous) @ app-main.bundle.js?28f670131887d2ef460e:2
(anonymous) @ app-main.bundle.js?28f670131887d2ef460e:2
(anonymous) @ redux.js:202
(anonymous) @ store.js:19
(anonymous) @ routes.jsx:175
u @ runtime.js:63
(anonymous) @ runtime.js:293
(anonymous) @ runtime.js:118
kMt @ app-main.bundle.js?28f670131887d2ef460e:2
a @ app-main.bundle.js?28f670131887d2ef460e:2
(anonymous) @ app-main.bundle.js?28f670131887d2ef460e:2
(anonymous) @ app-main.bundle.js?28f670131887d2ef460e:2
(anonymous) @ routes.jsx:174
o @ TransitionUtils.js:43
(anonymous) @ TransitionUtils.js:114
(anonymous) @ TransitionUtils.js:84
l @ AsyncUtils.js:34
g @ AsyncUtils.js:51
r @ TransitionUtils.js:83
runEnterHooks @ TransitionUtils.js:107
(anonymous) @ createTransitionManager.js:70
r @ TransitionUtils.js:74
runChangeHooks @ TransitionUtils.js:131
l @ createTransitionManager.js:67
(anonymous) @ createTransitionManager.js:45
l @ AsyncUtils.js:41
g @ AsyncUtils.js:51
F @ matchRoutes.js:231
c @ createTransitionManager.js:41
r @ createTransitionManager.js:222
listen @ createTransitionManager.js:246
componentWillMount @ Router.js:102
_o @ react-dom.production.min.js:137
$s @ react-dom.production.min.js:176
ya @ react-dom.production.min.js:263
mc @ react-dom.production.min.js:246
bc @ react-dom.production.min.js:246
cc @ react-dom.production.min.js:239
rc @ react-dom.production.min.js:230
Xc @ react-dom.production.min.js:281
(anonymous) @ react-dom.production.min.js:284
uc @ react-dom.production.min.js:240
Gc @ react-dom.production.min.js:284
t.render @ react-dom.production.min.js:290
oI @ app.js:78
(anonymous) @ app.js:111
react-dom.production.min.js:209 Error
    at _.$e (client:128:335)
    at new qp (client:226:3)
    at $p (client:245:34)
    at Mo (client:243:246)
    at Object.No [as renderButton] (client:215:62)
    at index.esm.js:89:82
    at la (react-dom.production.min.js:211:320)
    at _c (react-dom.production.min.js:257:303)
    at t.unstable_runWithPriority (scheduler.production.min.js:19:467)
    at Yi (react-dom.production.min.js:122:325)
oa @ react-dom.production.min.js:209
Oa.n.payload @ react-dom.production.min.js:227
bo @ react-dom.production.min.js:129
$s @ react-dom.production.min.js:179
ya @ react-dom.production.min.js:263
mc @ react-dom.production.min.js:246
bc @ react-dom.production.min.js:246
cc @ react-dom.production.min.js:239
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:19
Yi @ react-dom.production.min.js:122
Gi @ react-dom.production.min.js:123
Ki @ react-dom.production.min.js:122
_c @ react-dom.production.min.js:257
t.unstable_runWithPriority @ scheduler.production.min.js:19
Yi @ react-dom.production.min.js:122
wc @ react-dom.production.min.js:257
(anonymous) @ react-dom.production.min.js:256
N @ scheduler.production.min.js:17
w.port1.onmessage @ scheduler.production.min.js:14
app.js:16 Error: {componentStack: '\n    in l\n    in o\n    in div\n    in Styled(div)\n …\n    in p\n    in DragDropContextProvider\n    in I'}
f @ app.js:16
n.<computed> @ bindActionCreators.js:8
value @ ErrorBoundary.tsx:29
Oa.o.componentDidCatch.n.callback @ react-dom.production.min.js:227
go @ react-dom.production.min.js:131
ua @ react-dom.production.min.js:212
Oc @ react-dom.production.min.js:255
t.unstable_runWithPriority @ scheduler.production.min.js:19
Yi @ react-dom.production.min.js:122
zc @ react-dom.production.min.js:248
cc @ react-dom.production.min.js:239
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:19
Yi @ react-dom.production.min.js:122
Gi @ react-dom.production.min.js:123
Ki @ react-dom.production.min.js:122
_c @ react-dom.production.min.js:257
t.unstable_runWithPriority @ scheduler.production.min.js:19
Yi @ react-dom.production.min.js:122
wc @ react-dom.production.min.js:257
(anonymous) @ react-dom.production.min.js:256
N @ scheduler.production.min.js:17
w.port1.onmessage @ scheduler.production.min.js:14

```

### Information about your Metabase installation

```JSON
- My browser: Chrome/115.0.5790.102
- My OS: Ubuntu 22.04.2 LTS
- Metabase version: 0.46.6.1
- Metabase hosting environment: Kubernetes
- Metabase internal database: Postgres
```


### Severity

Blocking all users from Brazil to use Metabase

### Additional context

Problems like that have happened before #26184, and there are no changes we can make in Metabase to prevent it from happening again.

--- a/frontend/src/metabase/auth/components/GoogleButton/GoogleButton.tsx
+++ b/frontend/src/metabase/auth/components/GoogleButton/GoogleButton.tsx
@@ -3,6 +3,7 @@ import { t } from "ttag";
 import { getIn } from "icepick";
 import { GoogleOAuthProvider, GoogleLogin } from "@react-oauth/google";
 import { useDispatch, useSelector } from "metabase/lib/redux";
+import ErrorBoundary from "metabase/ErrorBoundary";
 import * as Urls from "metabase/lib/urls";
 import { loginGoogle } from "../../actions";
 import { getGoogleClientId, getSiteLocale } from "../../selectors";
@@ -49,15 +50,16 @@ export const GoogleButton = ({ redirectUrl, isCard }: GoogleButtonProps) => {
   return (
     <GoogleButtonRoot>
       {isCard && clientId ? (
-        <GoogleOAuthProvider clientId={clientId}>
-          <GoogleLogin
-            useOneTap
-            onSuccess={handleLogin}
-            onError={handleError}
-            locale={locale}
-            width="300"
-          />
-        </GoogleOAuthProvider>
+        <ErrorBoundary>
+          <GoogleOAuthProvider clientId={clientId}>
+            <GoogleLogin
+              useOneTap
+              onSuccess={handleLogin}
+              onError={handleError}
+              locale={locale}
+            />
+          </GoogleOAuthProvider>
+        </ErrorBoundary>
       ) : (
         <TextLink to={Urls.login(redirectUrl)}>
           {t`Sign in with Google`}

